apps/builder/app/canvas/features/webstudio-component/webstudio-component.tsx

```diff
@@ -108,12 +119,81 @@ type WebstudioComponentDevProps = {
   components: Components;
 };
 
-export const WebstudioComponentDev = ({
-  instance,
-  instanceSelector,
-  children,
-  components,
-}: WebstudioComponentDevProps) => {
+/**
+ * Radix's VisuallyHiddenPrimitive.Root https://github.com/radix-ui/primitives/blob/main/packages/react/visually-hidden/src/VisuallyHidden.tsx
+ * component makes content from hidden elements accessible to screen readers.
+ * Using the same Content children, however, leads to duplicated React elements, breaking our 'isSelected' logic.
+ * To prevent this, we check if an instance is a descendant of VisuallyHiddenPrimitive.Root, and if so,
+ * we avoid rendering it.
+ */
+const useIsScreenReaderDescendant = (ref: RefObject<HTMLElement>) => {
```

Comment:
could be ` useIsVisuallyHidden`

Reply:
because the logic isn't specifically about screenreader, screenreader is how its used, yes, but its not what it does